projects/admin/src/app/routes/base-route.ts

```diff
@@ -78,4 +80,17 @@ export class BaseRoute {
     }
     return data;
   }
+
+  /**
+   * Return the redirect url with an observable
+   * @param recordMetadata object
+   * @param baseUrl string
+   * @returns observable
+   */
+  protected redirectUrl(recordMetadata: any, baseUrl: string) {
+    const pid = ('$ref' in recordMetadata) ?
+      extractIdOnRef(recordMetadata.$ref) :
+      recordMetadata.pid;
```

Comment:
Maybe more readable (my point of view) : 
```
const pid = ('$ref' in recordMetadata) 
    ? extractIdOnRef(recordMetadata.$ref)
    : recordMetadata.pid;
```

Reply:
done